src/main/java/com/management/controller/CarController.java

```diff
@@ -18,27 +20,40 @@ public class CarController {
 	@Autowired
 	private CarRepository carRepository;
 
+	@Autowired
+	private CarService carService;
+
 	@GetMapping
-	public List<Car> getAllCars() {
+	public List<CarModel> getAllCars() {
 		return carRepository.findAll();
 	}
 
 	@PostMapping
-	public ResponseEntity<?> addCar(@RequestBody Car car) {
-		// Verifica se já existe um carro com a mesma marca, modelo e ano
-		Optional<Car> existingCar = carRepository.findByBrandAndModelAndYear(car.getBrand(), car.getModel(),
-				car.getYear());
-		if (existingCar.isPresent()) {
-			// Se já existir, retorna uma resposta de conflito
-			return ResponseEntity.status(HttpStatus.CONFLICT).body("Carro já incluído anteriormente.");
+	public ResponseEntity<?> addCar(@RequestBody CarModel carModel) {
+		if (carModel.getPlate() == null || carModel.getPlate().isEmpty()) {
+			return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("O campo 'plate' é obrigatório.");
+		}
+
+		try {
+			// Verifica se já existe um carro com a mesma marca, modelo e ano
+			Optional<CarModel> existingCar = carRepository.findByBrandAndModelAndYearAndPlate(carModel.getBrand(), carModel.getModel(),
+					carModel.getYear(), carModel.getPlate());
+			if (existingCar.isPresent()) {
+				// Se já existir, retorna uma resposta de conflito
+				return ResponseEntity.status(HttpStatus.CONFLICT).body("Carro já incluído anteriormente.");
+			}
+
+			// Se não existir, tenta salvar o novo carro e valida a placa
+			CarModel savedCar = carService.saveCar(carModel);
+			return ResponseEntity.ok(savedCar);
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

```suggestion
			return ResponseEntity.created(savedCar);
```


![image](https://github.com/herbertfilhik/car-management-api/assets/6626879/6b019fcb-1a73-4490-a719-b740f148b3d9)

Reply:
Done

Reply:
Eu acho que essa alteração nao foi comitada!